src/receive_imf.rs

```diff
@@ -747,17 +752,15 @@ async fn add_parts(
                         VerifiedEncryption::NotVerified(_) => ProtectionStatus::Unprotected,
                     };
 
-                    let chat = Chat::load_from_db(context, chat_id).await?;
-
+                    if chat.protected != ProtectionStatus::Unprotected
+                        && new_protection == ProtectionStatus::Unprotected
+                        // `chat.protected` must be maintained regardless of the
+                        // `Config::VerifiedOneOnOneChats` setting. That's why the check is here.
```

Comment:
```suggestion
                        // `chat.protected` must be maintained regardless of the `Config::VerifiedOneOnOneChats`.
                        // That's why the config is checked here, and not above.
```

I would find the comment easier to understand like this, though of course clarity is very subjective.